src/test/java/ch/unisg/ics/interactions/wot/td/io/IDGraphWriterTest.java

```diff
@@ -0,0 +1,185 @@
+
+  private Form POST_FORM;
+  private StringSchema STRING_SCHEMA;
+
+  private static final String IO_BASE_IRI = "http://example.org/";
+
+  @Before
+  public void init() {
+    this.POST_FORM = new Form.Builder("http://example.org/action")
+      .addOperationType(TD.invokeAction)
+      .setContentType("application/json")
+      .setMethodName("POST")
+      .build();
+
+    this.GET_FORM = new Form.Builder("http://example.org/action")
+      .addOperationType(TD.readProperty)
+      .setMethodName("GET")
+      .build();
+
+    this.STRING_SCHEMA  = new StringSchema.Builder().build();
+  }
+
+  @Test
+  public void testWriteCompleteID() throws IOException {
+    InteractionDescription completeIntD = InteractionDescription.builder()
+      .title(ID_TITLE)
+      .input(new InteractionInput("input1", POST_FORM, STRING_SCHEMA))
+      .output(new InteractionOutput("output1", STRING_SCHEMA))
+      .type(InteractionTypes.ACTION)
+      .uri("http://example.org/log#interaction-1")
+      .build();
+
+    String testID = PREFIXES + String.format(
+      "\n<http://example.org/log#interaction-1> a log:ActionLog;\n" +
+      "  td:title \"interaction-1\";\n" +
+      "  td:hasBase <http://example.org/log#interaction-1>;\n" +
+      "  log:created \"%s\";\n" +
+      "  td:hasInput [ a <https://www.w3.org/2019/wot/json-schema#StringSchema>;\n" +
+      "      td:value \"input1\";\n" +
+      "      td:hasForm [\n" +
+      "          htv:methodName \"POST\";\n" +
+      "          hctl:hasTarget <http://example.org/action>;\n" +
+      "          hctl:forContentType \"application/json\";\n" +
+      "          hctl:hasOperationType td:invokeAction\n" +
+      "        ]\n" +
+      "    ];\n" +
+      "  td:hasOutput [ a <https://www.w3.org/2019/wot/json-schema#StringSchema>;\n" +
+      "      td:value \"output1\"\n" +
+      "    ] .\n", getDateTimeWithoutSeconds());
+
+    assertIsomorphicGraphs(testID, completeIntD);
+  }
+
+  @Test
+  public void testWriteIDWithoutInput() throws IOException {
+    InteractionDescription intDWithoutInput = InteractionDescription.builder()
+      .title(ID_TITLE)
+      .type(InteractionTypes.EVENT)
+      .output(new InteractionOutput("output1", STRING_SCHEMA))
+      .uri("http://example.org/log#interaction-1")
+      .build();
+
+    String testID = PREFIXES + String.format(
+      "\n<http://example.org/log#interaction-1> a log:EventLog;\n" +
+        "  td:title \"interaction-1\";\n" +
+        "  td:hasBase <http://example.org/log#interaction-1>;\n" +
+        "  log:created \"%s\";\n" +
+        "  td:hasOutput [ a <https://www.w3.org/2019/wot/json-schema#StringSchema>;\n" +
+        "      td:value \"output1\"\n" +
+        "    ] .\n", getDateTimeWithoutSeconds());
+
+    assertIsomorphicGraphs(testID, intDWithoutInput);
+  }
+
+  @Test
+  public void testWriteIDWithoutInputValue() throws IOException {
+    InteractionDescription requestIntD = InteractionDescription.builder()
+      .title(ID_TITLE)
```

Comment:
Thank you very much, @flooji. Please, do not update the implementation based on the following. They are up to discussion:

I would propose making the interface (the method identifiers) more uniform to the existing library. E.g.:
`title` -> `setTitle` 
`input` -> `hasInput`
`uri` -> `setUri`
`type` -> What are `InteractionTypes` here?